openmdao/docs/features/model_visualization/n2_details.rst

```diff
@@ -0,0 +1,297 @@
+depending on whether you hover over a diagonal or an off-diagonal cell.
+
+When hovering over a diagonal item, the explicit connections to/from that variable are displayed. Red arrows indicate
+that element is the target of the connection. A green arrow indicates it is the source.
+
+When hovering over a off-diagonal, the behavior differs depending on whether the cell is in the lower or upper
+triangular part of the diagram.
+
+In the lower triangular, the marked cells represent upstream (in the sense of execution order of the
+components) connections. All the connections involved in that cycle
+are drawn to indicate which systems and components are involved in the convergence loop. Downstream connection arrows
+are green and upstream connections are red.
+
+In the upper triangular, the marked cells represent downstream connections. The red arrow that is drawn shows
+that connection.
+
+Clicking in the matrix on a connection or diagonal, lets us pin connections arrows so that they don't
+disappear when you hover away from the cell. You can clear these connections arrows by using the Clear Arrows and
+Connections button |clear_conn_arrows| in the toolbar.
+
+Solver Structure
+----------------
+Each System in a model can potentially have a linear and non-linear model associated with it. On the right side
+of the N2 diagram, the hierarchy of the solvers is shown. This time the hierarchy goes from right to left with the root
+of the model on the right.
+You can switch between showing the linear or non-linear
+solvers using the button |toggle_solver_names| in the toolbar at the top.
+
+The colors indicate the type of solver. The colors are shown in the Legend. The section of the legend showing
+the colors of the solver types changes depending on which are displayed in the solver structure.
+Here is what section of the legend looks like for both linear and nonlinear solvers.
+
+.. |nonlinear_solvers_legend| image:: images/nonlinear_solvers_legend.png
+   :align: top
+
+.. |linear_solvers_legend| image:: images/linear_solvers_legend.png
+   :align: top
+
+|linear_solvers_legend|    |nonlinear_solvers_legend|
+
+Toolbar
+-------
+
+The tool bar above the N2 diagram provides many useful capabilites.
+
+Zoomed Element
+**************
+The model hierarchy display has the notion of a `view`. The `view` is defined by the node in the hierarchy
+that is visible in the diagram. Initially, it is the root of the model but if you click on a subsystem, then
+only the part of the hierarchy beginning at that subsystem is shown.
+
+You can change the view and zoom into a System by clicking on it. Other parts of the hierarchy are no longer
+visible.
+
+The diagram keeps track of the view history and then lets us go back and forth through the history similar to undo and
+redo in apps.
+
+The buttons in the first group of the toolbar control the view.
+
+.. |return_to_root| image:: images/home.png
+   :align: middle
```

Comment:
Shouldn't all these images link to the actual icons we use, instead of ones that you added locally for this example? Maybe I don't fully understand how view_model html works, and where those come from, but it would be convenient to not have to modify them here if we modify the originals.

Reply:
That would be ideal! Unfortunately, the images in the N2 diagram are actually fonts. There's no image associated with them at least in our code. So I had to use this great website https://icomoon.io/app/#/select/image  to convert the fonts to PNG files. 

I also tried to see if there was a way to use the fonts in the rst files but didn't have luck with that. It got complicated.